frontend/src/features/side_window/beacon_malfunctions/BeaconMalfunctionDetailsFollowUp.tsx

```diff
@@ -0,0 +1,337 @@
+      } else {
+        nextNotificationsByDayAccumulated[dateWithoutTime] = [nextNotification]
+      }
+
+      return nextNotificationsByDayAccumulated
+    }, {}) || {}
+
+  const itemsByDate = mergeObjects(commentsByDate, mergeObjects(actionsByDate, notificationsByDate))
+
+  useEffect(() => {
+    if (comment?.length && textareaRef.current) {
+      const { scrollHeight } = textareaRef.current
+
+      if (scrollHeight > 50) {
+        textareaRef.current.style.height = `${scrollHeight}px`
+      } else {
+        textareaRef.current.style.height = `${50}px`
+      }
+    } else if (textareaRef.current) {
+      textareaRef.current.style.height = '50px'
+    }
+    setTextareaHeight(parseInt(textareaRef.current?.style?.height.replace('px', '') || '', 10))
+  }, [comment])
+
+  const saveComment = () => {
+    // @ts-ignore
+    dispatch(saveBeaconMalfunctionCommentFromKanban(beaconMalfunction?.id, comment)).then(() => {
+      setComment('')
+    })
+  }
+
+  const getFirstStatusAction = (_vesselStatus, malfunctionStartDateTime) => {
+    if (
+      _vesselStatus?.value === BeaconMalfunctionVesselStatus.AT_PORT ||
+      _vesselStatus?.value === BeaconMalfunctionVesselStatus.AT_SEA
+    ) {
+      return `Avarie ${_vesselStatus?.label?.replace(
+        'Navire ',
+        ''
+      )} ouverte dans MonitorFish, dernière émission à ${getTime(malfunctionStartDateTime, true)}`
+    }
+
+    if (_vesselStatus?.value === BeaconMalfunctionVesselStatus.NEVER_EMITTED) {
+      return 'Avarie ouverte dans MonitorFish, aucune émission du navire à ce jour.'
+    }
+
+    return ''
+  }
+
+  return (
+    <Body style={bodyStyle(smallSize)}>
+      <NumberComments style={numberCommentsStyle}>
+        <CommentsIcon style={commentsIconStyle} />
+        <NumberCommentsText
+          data-cy="side-window-beacon-malfunctions-detail-comments-number"
+          style={numberCommentsTextStyle}
+        >
+          {comments?.length} commentaire{comments?.length > 1 ? 's' : ''}
+        </NumberCommentsText>
+      </NumberComments>
+      <CommentsAndActions className="smooth-scroll" style={commentsAndActionsStyle(smallSize, textareaHeight || 0)}>
+        {firstStatus ? (
+          <BeaconMalfunctionDetailsFollowUpRow
+            dateText={getCommentOrActionDate(getDate(beaconMalfunction?.malfunctionStartDateTime))}
+            index={0}
+            smallSize={smallSize}
+          >
+            <BeaconMalfunctionDetailsFollowUpItem
+              contentText={getFirstStatusAction(vesselStatus, beaconMalfunction?.malfunctionStartDateTime)}
+              isLast
+              isLastDate={!itemsByDate?.length}
+              item={{
+                dateTime: beaconMalfunction?.malfunctionStartDateTime,
+                type: BeaconMalfunctionDetailsType.ACTION
+              }}
+              scrollToRef={scrollToRef}
+            />
+          </BeaconMalfunctionDetailsFollowUpRow>
+        ) : null}
+        {Object.keys(itemsByDate)
```

Comment:
We need a TODO to get rid of that (no transformation within the JSX + it can be done with lib utils).